_ml-commons-plugin/tutorials/semantic-search-byte-vectors.md

```diff
@@ -0,0 +1,314 @@
+        }
+    ]
+}
+```
+{% include copy-curl.html %}
+
+For compatibility with the Neural Search plugin, the `data_type` (output in the `inference_results.output.data_type` field of the response) must be set to `FLOAT32` in the post-processing function, even though the actual embedding type will be `INT8`.
+{: .important}
+
+Note the connector ID in the response; you'll use it to register the model.
+
+Register the model, providing its connector ID:
+
+```json
+POST /_plugins/_ml/models/_register?deploy=true
+{
+    "name": "Cohere embedding model for INT8 with search_document input type",
+    "function_name": "remote",
+    "description": "test model",
+    "connector_id": "your_connector_id"
+}
+```
+{% include copy-curl.html %}
+
+Note the model ID in the response; you'll use it in the following steps.
+
+Test the model, providing the model ID:
+
+```json
+POST /_plugins/_ml/models/your_embedding_model_id/_predict
+{
+    "parameters": {
+        "texts": ["hello", "goodbye"]
+    }
+}
+```
+{% include copy-curl.html %}
+
+The response contains inference results:
+
+```json
+{
+    "inference_results": [
+        {
+            "output": [
+                {
+                    "name": "sentence_embedding",
+                    "data_type": "FLOAT32",
+                    "shape": [
+                        1024
+                    ],
+                    "data": [
+                        20,
+                        -11,
+                        -60,
+                        -91,
+                        ...
+                    ]
+                },
+                {
+                    "name": "sentence_embedding",
+                    "data_type": "FLOAT32",
+                    "shape": [
+                        1024
+                    ],
+                    "data": [
+                        58,
+                        -30,
+                        9,
+                        -51,
+                        ...
+                    ]
+                }
+            ],
+            "status_code": 200
+        }
+    ]
+}
+```
+
+## Step 2: Ingest data
+
+First, create an ingest pipeline:
+
+```json
+PUT /_ingest/pipeline/pipeline-cohere
+{
+  "description": "Cohere embedding ingest pipeline",
+  "processors": [
+    {
+      "text_embedding": {
+        "model_id": "your_embedding_model_id_created_in_step1",
+        "field_map": {
+          "passage_text": "passage_embedding"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Next, create a k-NN index and set the `data_type` on the `passage_embedding` field to `byte` so it can hold quantized byte vectors:
```

Comment:
```suggestion
Next, create a k-NN index and set the `data_type` on the `passage_embedding` field to `byte` so it can hold byte-quantized vectors:
```